fcrepo4/fcrepo4
# Remove duplicate header information on TimeMap.

https://jira.duraspace.org/browse/FCREPO-2755

Remove duplicate header information on TimeMap.

# How should this be tested?
```
1. Create versioned RDFSource
$ curl -i -XPOST -H "Slug: VersionedResource" -H "Link: <http://mementoweb.org/ns#OriginalResource>; rel=\"type\"" "http://localhost:8080/rest" -H "Content-Type: text/n3" --data-binary "<> <info:fedora/test/field> \"Versioned Resource\""

2. Verify that there are no duplicate headers
$ curl -i http://localhost:8080/rest/VersionedResource/fcr:versions

HTTP/1.1 200 OK
Date: Wed, 18 Apr 2018 19:43:24 GMT
ETag: W/"6b4589605251b3eaa3760654e82489244bf3ad75"
Last-Modified: Wed, 18 Apr 2018 19:27:50 GMT
Link: <http://mementoweb.org/ns#TimeMap>; rel="type"
Link: <http://localhost:8080/rest/VersionedResource>; rel="original"
Vary-Post: Memento-Datetime
Allow: POST,HEAD,GET,OPTIONS
Link: <http://www.w3.org/ns/ldp#Resource>;rel="type"
Link: <http://www.w3.org/ns/ldp#RDFSource>;rel="type"
Preference-Applied: return=representation
Vary: Prefer
Vary: Accept
Vary: Range
Vary: Accept-Encoding
Vary: Accept-Language
Content-Type: text/turtle;charset=utf-8
Content-Length: 1503
Server: Jetty(9.2.3.v20140905)

3. Create versioned binary NonRdfSource
$ curl -i -XPOST -H "Slug: VersionedBinary" -H "Link: <http://mementoweb.org/ns#OriginalResource>; rel=\"type\"" http://localhost:8080/rest/ -H "Content-Type: application/octet-stream" --data-binary "Binary Content"

4. Verify that there are no duplicate headers in the binary TimeMap and its description TimeMap
$ curl -i http://localhost:8080/rest/VersionedBinary/fcr:versions

HTTP/1.1 200 OK
Date: Wed, 18 Apr 2018 19:19:52 GMT
Link: <http://mementoweb.org/ns#TimeMap>; rel="type"
Link: <http://localhost:8080/rest/VersionedBinary>; rel="original"
Vary-Post: Memento-Datetime
Allow: POST,HEAD,GET,OPTIONS
Link: <http://www.w3.org/ns/ldp#Resource>;rel="type"
Link: <http://www.w3.org/ns/ldp#RDFSource>;rel="type"
Preference-Applied: return=representation
Vary: Prefer
Vary: Accept
Vary: Range
Vary: Accept-Encoding
Vary: Accept-Language
Content-Type: text/turtle;charset=utf-8
Content-Length: 1170
Server: Jetty(9.2.3.v20140905)

$ curl -i http://localhost:8080/rest/VersionedBinary/fcr:metadata/fcr:versions

HTTP/1.1 200 OK
Date: Wed, 18 Apr 2018 19:26:09 GMT
ETag: W/"6e033f7ebfd5d2919894e9c05b8f5b5332169cf8"
Last-Modified: Wed, 18 Apr 2018 19:19:37 GMT
Link: <http://mementoweb.org/ns#TimeMap>; rel="type"
Link: <http://localhost:8080/rest/VersionedBinary/fcr:metadata>; rel="original"
Vary-Post: Memento-Datetime
Allow: POST,HEAD,GET,OPTIONS
Link: <http://www.w3.org/ns/ldp#Resource>;rel="type"
Link: <http://www.w3.org/ns/ldp#RDFSource>;rel="type"
Preference-Applied: return=representation
Vary: Prefer
Vary: Accept
Vary: Range
Vary: Accept-Encoding
Vary: Accept-Language
Content-Type: text/turtle;charset=utf-8
Content-Length: 1525
Server: Jetty(9.2.3.v20140905)
```
